working-w-copilot/copilot-in-projects-refactoring.md

```diff
@@ -0,0 +1,559 @@
+#  Copilot in Projects Pt. 2 - Refactoring
+
+## Goals
+
+We know how to generate code, start a chat, and work with Copilot's tools in VS Code, but what does it look like to apply what we've learned in a project? We'll mosey back to the [hello-books example project](https://github.com/AdaGold/hello-books-api) from Unit 2 to see how we can use Copilot to increase our productivity.
+
+Our goals for this lesson are to:
+- identify where Copilot can assist us with refactoring
+- get more experience adding and updating test cases with Copilot 
+
+## Refactoring with Copilot
+
+Writing new code with Copilot is great, but a significant portion of software development involves refactoring and updating existing code. We got a little taste of updating code with the `Book` class, but we're going to revisit a scenario that we've seen before – refactoring the `validate_book` function into a more generic function that we can use for any model. 
+
+For this section of the lesson, we're going to check out a different `hello-books` branch, [`07b-from-dict-refactor`](https://github.com/AdaGold/hello-books-api/tree/07b-from-dict-refactor), as the starting point.
+
+### !callout-info
+
+## Saving Project Changes
+
+If we forked the `hello-books` repo and want to keep any local changes that we made in the previous part of the lesson, we recommend pausing to create a new branch from the current one then pushing that code up before checking out the `07b-from-dict-refactor` branch. 
+
+### !end-callout
+
+### Planning the `validate_book` Refactor
+
+Our steps to plan a refactor with Copilot stay the same: we first need to identify our dependencies, then check if we have tests for code that will be affected. We will be refactoring the `validate_book` function in the `routes.py` file, so we should search the project for where the function is used to remind ourselves of our dependencies. 
+
+This is a scenario where Copilot shouldn't be used, since Copilot has access to open files, but not all of the files in your project. Indeed, if we ask Copilot for help, it will warn us about this lack of access and give us suggestions on how to search the project for dependencies.
+
+<details>
+  <summary>
+    If you're curious, feel free to try asking Copilot to identify dependencies for <code>`validate_book`</code> or expand this section to see our prompt and what Copilot suggested.
```

Comment:
- redundant non-functional code quotes
- removed you're to make this undirected
```suggestion
    If curious, feel free to try asking Copilot to identify dependencies for <code>validate_book</code> or expand this section to see our prompt and what Copilot suggested.
```